wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,306 @@
+from discord.ext import commands
+import discord
+from resources.utilities.embed import embed as em
+from WalleModels.models import BannedUsers, BanRecords
+import datetime
+import pytz
+import re
+from typing import Union
+import logging
+logger = logging.getLogger('wall_e')
+
+
+class Ban(commands.Cog):
+
+    def __init__(self, bot, config):
+        self.bot = bot
+        self.config = config
+        self.banlist = []
+        self.errorColour = 0xA6192E
+
+    @commands.Cog.listener(name='on_ready')
+    async def load(self):
+        """Grabs channel to send mod reports to and reads in the blacklist from db"""
+        report_channel = self.config.get_config_value('basic_config', 'COUNCIL_REPORT_CHANNEL')
+        logger.info(f'[Ban load()] Attempting to get the report channel: {report_channel}')
+
+        self.mod_channel = discord.utils.get(self.bot.guilds[0].channels, name=report_channel)
+        logger.info(f"[Ban info()] #{report_channel} channel {'successfully' if self.mod_channel else 'not'} found")
```

Comment:
you need to introduce some logic here to auto-create the report_channel for the staging platform.

you can look at https://github.com/CSSS/wall_e/blob/master/wall_e/src/resources/cogs/reminders.py#L205 for an idea of how to do it

also, what happens if the prod server doesn't have report_channel? yall rather that wall_e try to create it or just error out?

Reply:
wall_e runs on the csss guild and i dont see a situation where the channel doesn't exist

Reply:
I can. you and I leave wall_e development and none of the minions are aware that that channel_name can't be updated without being updated in wall_e

Reply:
good rule of thumb. just cause you can't think of when the situation might happen, doesn't mean it wont ever happen.

Reply:
for the test server we can set the env var to `council` cuz that channel already exists there

Reply:
is this a channel that wall_e only uses for sending messages or does it receive commands here too?

Reply:
if ur talking about council-summary in csss guild, then yea some times commands are called in there. buts its just mods testing out `.em` or `.warn` most of the time. 

if ur talking about council in the test server, no idea. i doubt anyone ever uses commands in there.

Reply:
then in that case, you will need to

1. in master
   1. ensure that wall_e checks to see if the channel reference by variable `report_channel` exists
   1. its up to you to decide if if doesn't exist, whether to
         1. exit wall_e or
         2. have wall_e create the channel
2. in staging: create a `report_channel` for each branch or PR that gets opened similar to how the bot_log_channel and bot_general_channels get created.

Reply:
i just add code that'll exit the bot if the channel doesnt exist.

Reply:
@modernNeo added the code for staging guild. and updated the dev.ini